# Examining App Development Potential Based on Data for Apps Available on both Apple and Android

In looking for datasets to dive into, I thought that these two from kaggle proved pretty interesting: https://www.kaggle.com/lava18/google-play-store-apps and https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps. 

These datasets had information from both google play and the app store regarding categories, ratings, genres, and pricing for all of the apps on each store.  If an app-making company was so inclined, it could use this information to gain market insight into the basis for a successful app.  

For this project, I decided to look at that data as if I was working for a company that builds free apps, with a main source of revenue being in-app ads.  This means that revenue would be gained from users engaging with those ads, which means that the more users using that app, the better.  How can I analyze this data to help gain insight into what type of free app would have the highest potential revenue source based on the features given?  Or - will the data reveal that a paid app might be more lucrative? Let's extract some info from this data and see what trends can be discovered! 

# Exploring the Initial Data

Let's see what we have at first glance. 

In [323]:
import pandas as pd
import numpy as np
from pandas import DataFrame

In [324]:
#Two files:

andr = pd.read_csv('/Users/atrain31/Desktop/Data/dq1/googleplaystore.csv')
apple = pd.read_csv('/Users/atrain31/Desktop/Data/dq1/AppleStore.csv')

In [325]:
andr.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [326]:
apple.head()

id               track_name  size_bytes currency  price  \
0  284882215                 Facebook   389879808      USD    0.0   
1  389801252                Instagram   113954816      USD    0.0   
2  529479190           Clash of Clans   116476928      USD    0.0   
3  420009108               Temple Run    65921024      USD    0.0   
4  284035177  Pandora - Music & Radio   130242560      USD    0.0   

   rating_count_tot  rating_count_ver  user_rating  user_rating_ver      ver  \
0           2974676               212          3.5              3.5     95.0   
1           2161558              1289          4.5              4.0    10.23   
2           2130805               579          4.5              4.5  9.24.12   
3           1724546              3842          4.5              4.0    1.6.2   
4           1126879              3594          4.0              4.5    8.4.1   

  cont_rating        prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
0          4+  Social Networking               37                1        29   
1         12+      Photo & Video               37                0        29   
2          9+              Games               38                5        18   
3          9+              Games               40                5         1   
4         12+              Music               37                4         1   

   vpp_lic  
0        1  
1        1  
2        1  
3        1  
4        1

In [327]:
andr.info(), apple.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
App               10841 non-null object
Category          10841 non-null object
Rating            9367 non-null float64
Reviews           10841 non-null object
Size              10841 non-null object
Installs          10841 non-null object
Type              10840 non-null object
Price             10841 non-null object
Content Rating    10840 non-null object
Genres            10841 non-null object
Last Updated      10841 non-null object
Current Ver       10833 non-null object
Android Ver       10838 non-null object
dtypes: float64(1), object(12)
memory usage: 1.1+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 16 columns):
id                  7197 non-null int64
track_name          7197 non-null object
size_bytes          7197 non-null int64
currency            7197 non-null object
price               7197 non-null float64
rating_

(None, None)

So there are some interesting possiblities in this data.  Listed here are the column names (Google play, followed by the Apple app store) for each dataframe that I feel can be correlated and grouped together for analysis, followed by why they may be useful: 

App/track_name - basic identifiers for each column (the apple 'id' doesn't seem to mean anything here)

Category/prime_genre - category of each app ('Genres' seems like it would be too specific and therefore give a cloudier picture of the genres).  This would most likely be the main determinant of what type of app to create - which categories are the most popular?  Which have future potential? 

Rating/user_rating - do higher ratings equal more downloads or possibly higher price? 

Reviews/rating_count_tot  (or Installs??)- see above

(Type & Price)/price - how does pricing affect downloads and ratings?

Content Rating/cont_rating

# Cleaning the Data & Initial Analysis

Before we start, some of the kaggle comments mentioned a line of data (#10472) that was incomplete.  I'm going to verify that the data isn't useful and drop that from the analysis.

In [328]:
print(andr.loc[[10472]])
play = andr.drop(10472)

                                           App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN  


## Duplicate and English Apps 

In [329]:
play['App'].nunique(), apple['track_name'].nunique()

(9659, 7195)

In [330]:
play_dup = len(play) - play['App'].nunique()
apple_dup = len(apple) - apple['track_name'].nunique()
play_dup, apple_dup

(1181, 2)

Looks the like google play has 1181 duplicate app names, while the app store has two.  Let's create a dictionary to make sure that we have only one of each app in our data; we'll base which one to keep on which one has the most reviews. 

In [331]:
rev_max_play = {}

for i,row in play.iterrows():
    app = row['App']
    num_rev = float(row['Reviews'])
    
    if app in rev_max_play and rev_max_play[app] < num_rev:
        rev_max_play[app] = num_rev
        
    elif app not in rev_max_play:
        rev_max_play[app] = num_rev

len(rev_max_play)

9659

In [332]:
rev_max_apple = {}

for i,row in apple.iterrows():
    appl = row['track_name']
    num_revi = float(row['rating_count_tot'])
    
    if app in rev_max_apple and rev_max_apple[appl] < num_revi:
        rev_max_apple[appl] = num_revi
        
    elif app not in rev_max_apple:
        rev_max_apple[appl] = num_revi

len(rev_max_apple)

7195

In [333]:
play_clean = []
already_added = []

for i,app in play.iterrows():
    name = app['App']
    n_reviews = float(app['Reviews'])
    
    if (rev_max_play[name] == n_reviews) and (name not in already_added):
        play_clean.append(app)
        already_added.append(name) 

In [334]:
apple_clean = []
already_added2 = []

for i,app in apple.iterrows():
    name = app['track_name']
    n_reviews = float(app['rating_count_tot'])
    
    if (rev_max_apple[name] == n_reviews) and (name not in already_added2):
        apple_clean.append(app)
        already_added2.append(name) 

In [335]:
len(play_clean), len(apple_clean)

(9659, 7195)

In [336]:
play_clean_df = pd.DataFrame(play_clean)
apple_clean_df = pd.DataFrame(apple_clean)

apple_clean_df.to_csv('/Users/atrain31/Desktop/Data/dq1/apple_clean.csv')
play_clean_df.to_csv('/Users/atrain31/Desktop/Data/dq1/play_clean.csv')

Now we have two lists, play_clean and apple_clean, which have only one of each app in the data.  The dictionaries worked as the cleaned lists are the exact same length as the number of unique values we originally found in the first step above. 

Further exploring the data, it would most likely be difficult to analyze data from non-english apps.  I'm going to weed those out. 

In [337]:
def english_app(string):
    non_english_count = 0
    for character in string:
        number = ord(character)
        if number > 127:
            non_english_count += 1
        else:
            non_english_count += 0

    if non_english_count>2:
        return False
    else:
        return True
    
#Testing the function:

english_app('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [338]:
play_english = []
for app in play_clean:
    name = app[1]
    if english_app(name):
        play_english.append(app)
    else:
        None

apple_english = []
for app in apple_clean:
    name = app[1]
    if english_app(name):
        apple_english.append(app)
    else:
        None
len(play_english),len(apple_english)    

(9659, 6153)

We are now down to 9659 apps on google play and 6152 on the app store. 

Let's see how many of those apps were free - remember,  in the scenario I mentioned in the intro, I'm looking for free apps in an effort to develop an app based on ad revenue.  

## Pricing

Let's look at the pricing of the available apps:

In [339]:
play_english_df = pd.DataFrame(play_english)
apple_english_df = pd.DataFrame(apple_english)

In [340]:
play_english_df.Type.unique()

array(['Free', 'Paid', nan], dtype=object)

Nan comes up as a type of app, so let's see how many are listed: 

In [341]:
play_english_df['Type'].isnull().sum()

1

In [342]:
play2 = play_english_df[play_english_df['Type'].isnull()]
play2

App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

The price of the Type NaN app is zero and there is only one instance of this, but let's base our pricing on the "Price" column rather than the "Type" column and go from there (also the app store data does not have a Type column).  Let's see what we have for a ranger for both datasets:

In [343]:
play_english_df['Price'].isnull().sum(), apple_english_df['price'].isnull().sum()

(0, 0)

In [344]:
apple_english_df['price'].min(), apple_english_df['price'].max()

(0.0, 299.99)

In [345]:
play_english_df['Price'].min(), play_english_df['Price'].max()

('$0.99', '0')

The Apple range looks good, but the Google range is off.  Let's fix that: 

In [346]:
play_english_df['Price'] = play_english_df['Price'].str.replace('$','')

In [347]:
play_english_df['Price'] = pd.to_numeric(play_english_df['Price'])

In [348]:
play_english_df['Price'].min(), play_english_df['Price'].max()

(0.0, 400.0)

In [349]:
apple_english_df.to_csv('/Users/atrain31/Desktop/Data/dq1/apple_english.csv')
play_english_df.to_csv('/Users/atrain31/Desktop/Data/dq1/play_english.csv')

Before moving forward, I'd like to take a look at how prevalent paid apps are vs free apps and get an idea of how popular each type is with users. 

In [350]:
play_free = play_english_df[play_english_df['Price']==0.0]
apple_free = apple_english_df[apple_english_df['price']==0.0]

In [351]:
len(play_free), len(apple_free)

(8905, 3201)

There are 9659 total unique English apps in the Play store, 8905 of which are free; 6153 and 3201 in the app store.  I'd like to plug this into Tableau and see what it looks like:

In [352]:
%%HTML
<div class='tableauPlaceholder' id='viz1597409129902' style='position: relative'><noscript><a href='#'><img alt=' ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppPriceRelativetoNumberofReviewsandAverageRating&#47;Sheet1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='GooglePlayAppPriceRelativetoNumberofReviewsandAverageRating&#47;Sheet1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppPriceRelativetoNumberofReviewsandAverageRating&#47;Sheet1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1597409129902');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

The extreme popularity of ~50 apps totally skewers this graph, making it almost unusable, BUT - one thing we can see is that all of the very popular apps are free apps.  Let's further examine this: 

In [354]:
play_paid = play_english_df[play_english_df['Price']>0.0]

play_free['Reviews'] = play_free['Reviews'].astype(int)
play_paid['Reviews'] = play_paid['Reviews'].astype(int)

play_paid_popular = play_paid[play_paid['Reviews']>500000]
play_free_popular = play_free[play_free['Reviews']>500000]
len(play_paid_popular), len(play_free_popular)

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until
//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  after removing the cwd from sys.path.


(1, 585)

In [355]:
play_paid_popular2 = play_paid[play_paid['Reviews']>100000]
play_free_popular2 = play_free[play_free['Reviews']>100000]
len(play_paid_popular2), len(play_free_popular2)

(8, 1443)

In [356]:
print(play_free['Reviews'].max())
print(play_paid['Reviews'].max())

78158306
2376564


Wow!  This is a huge difference in popularity.  Not only do the most popular apps get tens of millions of reviews, but with the exception of only one, there are no paid apps with over 500,000 reviews.  And - only eight paid apps surpass 100,000 reviews, while 1443(!) surpass this amount.  

Let's look at the stats for the app store:

In [357]:
apple_paid = apple_english_df[apple_english_df['price']>0.0]

apple_free['rating_count_tot'] = apple_free['rating_count_tot'].astype(int)

apple_paid_popular = apple_paid[apple_paid['rating_count_tot']>100000]
apple_free_popular = apple_free[apple_free['rating_count_tot']>100000]
len(apple_paid_popular), len(apple_free_popular)

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


(19, 199)

Again, there is a large difference, almost ten fold in the popularity of free apps vs. paid apps

Seeing as we're looking into an app with the highest earning potential, it would seem that our chances would be much better with a free app with revenue earned through advertisements.  

For the remainder of this analysis, I'll base our data on the free apps only. 

## Categories 

In [358]:
play_free['Category'].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION'],
      dtype=object)

In [359]:
apple_free['prime_genre'].unique()

array(['Social Networking', 'Photo & Video', 'Games', 'Music',
       'Reference', 'Health & Fitness', 'Weather', 'Utilities', 'Travel',
       'Shopping', 'News', 'Navigation', 'Lifestyle', 'Entertainment',
       'Food & Drink', 'Sports', 'Book', 'Finance', 'Education',
       'Productivity', 'Business', 'Catalogs', 'Medical'], dtype=object)

The differences in categories between the two operating systems prevents us really from combining the data together. 

## Number of Reviews(Ratings)/Installs 

In [360]:
#Cleaning the Installs column:
play_free['Installs'] = play_free['Installs'].str.replace('+','')
play_free['Installs'] = play_free['Installs'].str.replace(',','')
play_free['Installs'] = play_free['Installs'].astype(int)

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.or

In [374]:
play_free['Installs'].unique()

array([     10000,    5000000,   50000000,     100000,      50000,
          1000000,   10000000,       5000,     500000, 1000000000,
        100000000,       1000,  500000000,        500,        100,
               50,         10,          1,          5,          0])

In [393]:
play_free.sort_values('Installs',ascending=False)

App             Category  Rating  \
336                          WhatsApp Messenger        COMMUNICATION     4.4   
391               Skype - free IM & video calls        COMMUNICATION     4.1   
2604                                  Instagram               SOCIAL     4.5   
451                                       Gmail        COMMUNICATION     4.3   
464                                    Hangouts        COMMUNICATION     4.0   
3665                                    YouTube        VIDEO_PLAYERS     4.3   
2554                                    Google+               SOCIAL     4.2   
1872                             Subway Surfers                 GAME     4.5   
2884                              Google Photos          PHOTOGRAPHY     4.5   
3523                               Google Drive         PRODUCTIVITY     4.4   
411                Google Chrome: Fast & Secure        COMMUNICATION     4.3   
152                           Google Play Books  BOOKS_AND_REFERENCE     3.9   
2544                                   Facebook               SOCIAL     4.1   
3234                                     Google                TOOLS     4.4   
3232                         Google Street View     TRAVEL_AND_LOCAL     4.2   
3223                  Maps - Navigate & Explore     TRAVEL_AND_LOCAL     4.3   
5856                          Google Play Games               FAMILY     4.3   
3687                    Google Play Movies & TV        VIDEO_PLAYERS     3.7   
9844                                Google News   NEWS_AND_MAGAZINES     3.9   
382    Messenger – Text and Video Chat for Free        COMMUNICATION     4.0   
383               imo free video calls and chat        COMMUNICATION     4.3   
3739                                    Twitter   NEWS_AND_MAGAZINES     4.3   
3819               Flipboard: News For Our Time   NEWS_AND_MAGAZINES     4.4   
403                 LINE: Free Calls & Messages        COMMUNICATION     4.2   
3450                             Microsoft Word         PRODUCTIVITY     4.5   
3255                 SHAREit - Transfer & Share                TOOLS     4.6   
3265               Gboard - the Google Keyboard                TOOLS     4.2   
4005    Clean Master- Space Cleaner & Antivirus                TOOLS     4.7   
3711                                  MX Player        VIDEO_PLAYERS     4.5   
1869                           Candy Crush Saga                 GAME     4.4   
...                                         ...                  ...     ...   
10745                                   FP Boss              FINANCE     NaN   
10701                                 F-O-Meter               FAMILY     NaN   
4550          RMEduS - 음성인식을 활용한 R 프로그래밍 실습 시스템               FAMILY     NaN   
9262                             EC-Contractors       HOUSE_AND_HOME     NaN   
9264                                 EC Sampler         PRODUCTIVITY     NaN   
8509                             Dr D K Olukoya            LIFESTYLE     NaN   
10654               Wallpapers FN Herstal FNP 9      PERSONALIZATION     NaN   
5857                                   Ay Peruk             BUSINESS     NaN   
9139                                DZ Register         PRODUCTIVITY     NaN   
5160                                  Kay AH GA              MEDICAL     NaN   
10417                                 FH Wallet              FINANCE     NaN   
10299                                FD CANNECT                TOOLS     NaN   
10101                                 Amleen Ey               SOCIAL     NaN   
2454                        KBA-EZ Health Guide              MEDICAL     5.0   
9453                             EJ Trívia Game               FAMILY     NaN   
10775                        FQ - Football Quiz               SPORTS     NaN   
7522            CL 2ne1 Wallpaper KPOP Fans All      PERSONALIZATION     NaN   
5837     Ako ay may lobo Pinoy Kid Song Offline               FAMILY     NaN   
7521            New CL 2ne1 Wallpaper KPOP l

Here we can see that there are several dominant apps in various categories that are outliers and don't represent 'regular' apps. This is better illustrated below:

In [405]:
%%HTML
<div class='tableauPlaceholder' id='viz1597608679877' style='position: relative'><noscript><a href='#'><img alt=' ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayOutliers&#47;Sheet4&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='GooglePlayOutliers&#47;Sheet4' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayOutliers&#47;Sheet4&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1597608679877');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Let's see what happens when we sort by the mean # of installs for each category:

In [406]:
play_free.groupby('Category')['Installs'].mean().sort_values(ascending=False).reset_index()

Category      Installs
0         COMMUNICATION  3.832263e+07
1         VIDEO_PLAYERS  2.457395e+07
2                SOCIAL  2.325365e+07
3           PHOTOGRAPHY  1.777202e+07
4          PRODUCTIVITY  1.673896e+07
5                  GAME  1.555200e+07
6      TRAVEL_AND_LOCAL  1.398408e+07
7         ENTERTAINMENT  1.164071e+07
8                 TOOLS  1.078701e+07
9    NEWS_AND_MAGAZINES  9.401636e+06
10  BOOKS_AND_REFERENCE  8.587352e+06
11             SHOPPING  7.001693e+06
12      PERSONALIZATION  5.183851e+06
13              WEATHER  5.074486e+06
14   HEALTH_AND_FITNESS  4.188822e+06
15  MAPS_AND_NAVIGATION  3.993340e+06
16               FAMILY  3.668871e+06
17               SPORTS  3.638640e+06
18       ART_AND_DESIGN  1.952105e+06
19       FOOD_AND_DRINK  1.924898e+06
20            EDUCATION  1.825481e+06
21             BUSINESS  1.708216e+06
22            LIFESTYLE  1.436127e+06
23              FINANCE  1.387692e+06
24       HOUSE_AND_HOME  1.331541e+06
25               DATING  8.540288e+05
26               COMICS  8.032348e+05
27    AUTO_AND_VEHICLES  6.473178e+05
28   LIBRARIES_AND_DEMO  6.385037e+05
29            PARENTING  5.426036e+05
30               BEAUTY  5.131519e+05
31               EVENTS  2.535422e+05
32              MEDICAL  1.205506e+05

Now let's see what happens when we take out the top 5% of installed apps and the bottom 5% and we revisit the mean, we can see that the the order has been displaced once the extremely popular apps are removed.  I'd probably stay away from the video players, social, and communication categories as making an app in those categories would face stiff competition.  

In [407]:
iqr = play_free[play_free['Installs'].between(play_free['Installs'].quantile(.05), play_free['Installs'].quantile(.95), inclusive=True)]
iqr.groupby('Category')['Installs'].mean().sort_values(ascending=False).reset_index()

Category      Installs
0         ENTERTAINMENT  3.808684e+06
1           PHOTOGRAPHY  3.453283e+06
2                  GAME  3.306546e+06
3              SHOPPING  3.074517e+06
4         COMMUNICATION  2.476439e+06
5               WEATHER  2.465977e+06
6         VIDEO_PLAYERS  2.387278e+06
7                SOCIAL  2.146381e+06
8                SPORTS  2.081032e+06
9      TRAVEL_AND_LOCAL  2.055750e+06
10       FOOD_AND_DRINK  2.055716e+06
11   HEALTH_AND_FITNESS  1.958525e+06
12         PRODUCTIVITY  1.887384e+06
13            EDUCATION  1.825481e+06
14      PERSONALIZATION  1.808437e+06
15  MAPS_AND_NAVIGATION  1.721701e+06
16                TOOLS  1.715416e+06
17               FAMILY  1.481028e+06
18  BOOKS_AND_REFERENCE  1.469316e+06
19       HOUSE_AND_HOME  1.369049e+06
20   NEWS_AND_MAGAZINES  1.358350e+06
21       ART_AND_DESIGN  1.128966e+06
22            LIFESTYLE  1.081731e+06
23              FINANCE  9.506639e+05
24               DATING  9.270701e+05
25               COMICS  8.032348e+05
26             BUSINESS  7.306260e+05
27    AUTO_AND_VEHICLES  7.172973e+05
28   LIBRARIES_AND_DEMO  6.462902e+05
29            PARENTING  5.521228e+05
30               BEAUTY  5.131519e+05
31               EVENTS  2.802298e+05
32              MEDICAL  1.367103e+05

In [385]:
len(iqr), len(play_free)

(8032, 8905)

In [394]:
iqr.sort_values('Installs',ascending=False)

App            Category  \
5948                              PBA® Bowling Challenge              SPORTS   
3149                     Despegar.com Hotels and Flights    TRAVEL_AND_LOCAL   
3268                           Google app for Android TV               TOOLS   
3267                                    AT&T Smart Wi-Fi               TOOLS   
3264                                     HTC Lock Screen               TOOLS   
3263                                               Phone               TOOLS   
3262                                               Clear               TOOLS   
3260   File Manager -- Take Command of Your Files Easily               TOOLS   
3257     Files Go by Google: Free up space on your phone               TOOLS   
3256                       ZenUI Keyboard – Emoji, Theme               TOOLS   
3253                                          ZenUI Help               TOOLS   
3251     Samsung Max - Data Savings & Privacy Protection               TOOLS   
3249                                 ASUS Sound Recorder               TOOLS   
3248   Calculator - free calculator, multi calculator...               TOOLS   
3245                                             myMetro               TOOLS   
3241                                          Moto Voice               TOOLS   
3236                                        Moto Display               TOOLS   
3224                                GPS Status & Toolbox    TRAVEL_AND_LOCAL   
3217                         Agoda – Hotel Booking Deals    TRAVEL_AND_LOCAL   
3215                                              Airbnb    TRAVEL_AND_LOCAL   
3210   Hotels.com: Book Hotel Rooms & Find Vacation D...    TRAVEL_AND_LOCAL   
3194                                 easyJet: Travel App    TRAVEL_AND_LOCAL   
3169               Yelp: Food, Shopping, Services Nearby    TRAVEL_AND_LOCAL   
3163   MakeMyTrip-Flight Hotel Bus Cab IRCTC Rail Boo...    TRAVEL_AND_LOCAL   
3162     Where is my Train : Indian Railway & PNR Status    TRAVEL_AND_LOCAL   
3269       Sound Recorder: Recorder & Voice Changer Free               TOOLS   
3270                             Remote Link (PC Remote)               TOOLS   
3271                                     HTC Sense Input               TOOLS   
3320        OpenVPN Connect – Fast & Safe SSL VPN Client               TOOLS   
3347                               AppLock - Fingerprint               TOOLS   
...                                                  ...                 ...   
8577                                      DM AirDisk NAS               TOOLS   
7188                                       CD Padel Mora              SPORTS   
7322                                          music (CG)              FAMILY   
9994                                             EW Gate            BUSINESS   
5151                                       Clarksburg AH             MEDICAL   
6605                                               MI-BP  HEALTH_AND_FITNESS   
10369  FG. Chic - House Lounge Musique - Radios de Fr...              FAMILY   
9180                               Debra Care Conference            BUSINESS   
8278                                      Pepsi Cards DC              COMICS   
5134                                         Woodland AH             MEDICAL   
5136                                        Angelcare AH             MEDICAL   
9319                                    EG Mantenimiento            BUSINESS   
5140                                           Sayers AH             MEDICAL   
7888                                          CT CONNECT        PRODUCTIVITY   
6828                                     BU Beauty Rooms              BEAUTY   
6544                                          BN DB1 App        PRODUCTIVITY   
10349                                    Santa Fe Thrive  HEALTH_AND_FITNESS   
5730                                        AV Anthony's      FOOD_AND_DRINK   
6832                                          Bu Nedi

In [386]:
play_free2 = iqr

I think that the number of times an app was reviewed is a nice gauge to see how popular an app was (it's also necessary when looking at the Apple data as there is no data on the amount of installs). On Google it is the 'Reviews' column and on Apple it is the 'rating_count_tot' column.  I want to make sure the results aren't skewed with apps that were rated a limited number of times, so I'm going to omit the lowest quartile when looking at the number of reviews/ratings.  

In [387]:
play_free2['Reviews'] = play_free2['Reviews'].astype(int)
play_free2.Reviews.quantile([0.25,0.5,0.75])

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


0.25       42.00
0.50     1375.50
0.75    27506.75
Name: Reviews, dtype: float64

In [389]:
play_english = play_free2[play_free2['Reviews']>42]

In [431]:
play_english.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                         Paper flowers instructions  ART_AND_DESIGN     4.4   
6            Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   

   Reviews  Size  Installs  Type  Price Content Rating  \
0      159   19M     10000  Free    0.0       Everyone   
2    87510  8.7M   5000000  Free    0.0       Everyone   
4      967  2.8M    100000  Free    0.0       Everyone   
5      167  5.6M     50000  Free    0.0       Everyone   
6      178   19M     50000  Free    0.0       Everyone   

                    Genres     Last Updated Current Ver   Android Ver  
0             Art & Design  January 7, 2018       1.0.0  4.0.3 and up  
2             Art & Design   August 1, 2018       1.2.4  4.0.3 and up  
4  Art & Design;Creativity    June 20, 2018         1.1    4.4 and up  
5             Art & Design   March 26, 2017         1.0    2.3 and up  
6             Art & Design   April 26, 2018         1.1  4.0.3 and up

In [363]:
apple_free['rating_count_tot'].min(), apple_free['rating_count_tot'].max()

(0, 2974676)

In [408]:
apple_free.rating_count_tot.quantile([0.25,0.5,0.75,0.95])

0.25       147.0
0.50      1204.0
0.75      9975.0
0.95    119316.0
Name: rating_count_tot, dtype: float64

In [412]:
apple_english = apple_free[(apple_free['rating_count_tot']>147) & (apple_free['rating_count_tot']<119316)]

Again, let's look at the before and after when the top and bottom 5% are removed:

In [426]:
apple_free.groupby('prime_genre')['rating_count_tot'].mean().sort_values(ascending=False).reset_index()

prime_genre  rating_count_tot
0          Navigation      86090.333333
1           Reference      79350.470588
2   Social Networking      71548.349057
3               Music      57326.530303
4             Weather      52279.892857
5                Book      46384.916667
6        Food & Drink      33333.923077
7             Finance      32367.028571
8       Photo & Video      28441.543750
9              Travel      28243.800000
10           Shopping      27230.734940
11   Health & Fitness      23298.015385
12             Sports      23008.898551
13              Games      22910.507511
14               News      21248.023256
15       Productivity      21028.410714
16          Utilities      19156.493671
17          Lifestyle      16815.480000
18      Entertainment      14195.358566
19           Business       7491.117647
20          Education       7003.983051
21           Catalogs       4004.000000
22            Medical        612.000000

In [427]:
apple_english.groupby('prime_genre')['rating_count_tot'].mean().sort_values(ascending=False).reset_index()

prime_genre  rating_count_tot
0                Book      38066.750000
1             Weather      27613.500000
2             Finance      19415.363636
3            Shopping      18160.338462
4        Food & Drink      17887.647059
5   Social Networking      16858.388235
6        Productivity      16823.666667
7              Sports      15323.162791
8                News      15225.785714
9           Utilities      14104.490909
10              Music      13622.705882
11          Reference      13580.750000
12      Photo & Video      12719.256198
13      Entertainment      12341.701571
14   Health & Fitness      11256.954545
15              Games      11107.454474
16             Travel      10463.566667
17          Lifestyle       9322.710526
18           Business       9088.928571
19          Education       7183.453333
20         Navigation       5526.666667
21           Catalogs       4004.000000
22            Medical       1214.333333

In [366]:
#DL these clean files to desktop for Tableau use 

apple_english.to_csv('/Users/atrain31/Desktop/Data/dq1/apple_english.csv')
play_english.to_csv('/Users/atrain31/Desktop/Data/dq1/play_english.csv')

## Ratings

In [367]:
%%HTML

<div class='tableauPlaceholder' id='viz1597420646361' style='position: relative'><noscript><a href='#'><img alt=' ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppRatingvs_Installs&#47;Sheet2&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='GooglePlayAppRatingvs_Installs&#47;Sheet2' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppRatingvs_Installs&#47;Sheet2&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1597420646361');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Conventional wisdom would point to app ratings increasing as the number of installs increases, and this chart reflects that thought.  Obviously a poorly reviewed product will decrease it's popularity.  Going forward we can assume that a well-made and well-reviewed app will translate into higher downloads. 

# Further Analysis

Now that we have the list of apps that are both in english and free, we can really dive into some potential trends.  I'd like to check on how many reviews per category are present, so I'll create two dictionaries for each dataset: 1) the count for the number of apps for each category and 2) the number of reviews for each category.  

In [419]:
# Google Play
categories_two  = list(set(play_english['Category']))
categ_inst_dict = {}
categ_count = {}
categ_avg_rating = {}

for cat in categories_two:

    new_df = play_english[play_english['Category']==cat]
    
    count = 0
    count += len(new_df)
    sum_of_installs = new_df['Reviews'].sum()
    categ_inst_dict[cat] = sum_of_installs
    categ_count[cat] = count
    
    ratings_count = 0
    ratings_df = new_df.dropna(subset=['Rating'])
    ratings_count = len(ratings_df)
    categ_avg_rating[cat] = (ratings_df['Rating'].sum())/ratings_count

    
#App Store
categories_three  = list(set(apple_english['prime_genre']))
a_categ_inst_dict = {}
a_categ_count = {}

for cat in categories_three:
    a_count = 0
    a_new_df = apple_english[apple_english['prime_genre']==cat]
    a_new_df['rating_count_tot'] = a_new_df['rating_count_tot'].astype(int)
    a_count += len(a_new_df)
    a_sum_of_installs = a_new_df['rating_count_tot'].sum()
    a_categ_inst_dict[cat] = a_sum_of_installs
    a_categ_count[cat] = a_count

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:31: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [420]:
# Testing the dictionary
categ_inst_dict['BUSINESS'], categ_count['BUSINESS'], categ_avg_rating['BUSINESS']

(4931593, 164, 4.093827160493827)

The Google data: 

In [421]:
new_play_df = pd.DataFrame(categories_two,columns=['Category'])
new_play_df['Count'] = new_play_df['Category'].map(categ_count)
new_play_df['# of Reviews'] = new_play_df['Category'].map(categ_inst_dict)
new_play_df['Avg # Reviews Per App'] = (new_play_df['# of Reviews'] / new_play_df['Count'])
new_play_df['% of Total Apps'] = (new_play_df['Count'] / new_play_df['Count'].sum())
new_play_df.sort_values('% of Total Apps',ascending=False)

Category  Count  # of Reviews  Avg # Reviews Per App  \
20               FAMILY   1172      81417526           69468.878840   
8                  GAME    669     107215876          160262.893871   
18                TOOLS    488      26208389           53705.715164   
26              FINANCE    243      11631014           47864.255144   
23            LIFESTYLE    214       8912231           41645.939252   
4                SPORTS    209      19403576           92840.076555   
2          PRODUCTIVITY    205      13842509           67524.434146   
7    HEALTH_AND_FITNESS    195      14406257           73878.241026   
11      PERSONALIZATION    187      11222479           60013.256684   
5           PHOTOGRAPHY    180      18485694          102698.300000   
1    NEWS_AND_MAGAZINES    173       8352018           48277.560694   
28        COMMUNICATION    166      15789565           95117.861446   
0              BUSINESS    164       4931593           30070.689024   
30               SOCIAL    154      17539773          113894.629870   
21             SHOPPING    150      11677111           77847.406667   
6      TRAVEL_AND_LOCAL    148       8167726           55187.337838   
13              MEDICAL    135       1166193            8638.466667   
31  BOOKS_AND_REFERENCE    130       8548324           65756.338462   
32        VIDEO_PLAYERS    109       9703585           89023.715596   
9                DATING    109       3621923           33228.651376   
19            EDUCATION    104       5854254           56290.903846   
14  MAPS_AND_NAVIGATION     92       3520106           38262.021739   
12       FOOD_AND_DRINK     87       6322480           72672.183908   
16        ENTERTAINMENT     76       7482110           98448.815789   
25       HOUSE_AND_HOME     61       1929707           31634.540984   
3    LIBRARIES_AND_DEMO     61        906493           14860.540984   
15              WEATHER     60       4752509           79208.483333   
22    AUTO_AND_VEHICLES     59       1159320           19649.491525   
29               COMICS     50       2342284           46845.680000   
17       ART_AND_DESIGN     49       1192103           24328.632653   
27            PARENTING     48        949667           19784.729167   
24               BEAUTY     37        395989           10702.405405   
10               EVENTS     33        160777            4872.030303   

    % of Total Apps  
20         0.194781  
8          0.111185  
18         0.081104  
26         0.040386  
23         0.035566  
4          0.034735  
2          0.034070  
7          0.032408  
11         0.031079  
5          0.029915  
1          0.028752  
28         0.027588  
0          0.027256  
30         0.025594  
21         0.024929  
6          0.024597  
13         0.022436  
31         0.021605  
32         0.018115  
9          0.018115  
19         0.017284  
14         0.015290  
12         0.014459  
16         0.012631  
25         0.010138  
3          0.010138  
15         0.009972  
22         0.009806  
29         0.008310  
17         0.008144  
27         0.007977  
24         0.006149  
10         0.005484

In [422]:
new_play_df.sort_values('Avg # Reviews Per App',ascending=False)

Category  Count  # of Reviews  Avg # Reviews Per App  \
8                  GAME    669     107215876          160262.893871   
30               SOCIAL    154      17539773          113894.629870   
5           PHOTOGRAPHY    180      18485694          102698.300000   
16        ENTERTAINMENT     76       7482110           98448.815789   
28        COMMUNICATION    166      15789565           95117.861446   
4                SPORTS    209      19403576           92840.076555   
32        VIDEO_PLAYERS    109       9703585           89023.715596   
15              WEATHER     60       4752509           79208.483333   
21             SHOPPING    150      11677111           77847.406667   
7    HEALTH_AND_FITNESS    195      14406257           73878.241026   
12       FOOD_AND_DRINK     87       6322480           72672.183908   
20               FAMILY   1172      81417526           69468.878840   
2          PRODUCTIVITY    205      13842509           67524.434146   
31  BOOKS_AND_REFERENCE    130       8548324           65756.338462   
11      PERSONALIZATION    187      11222479           60013.256684   
19            EDUCATION    104       5854254           56290.903846   
6      TRAVEL_AND_LOCAL    148       8167726           55187.337838   
18                TOOLS    488      26208389           53705.715164   
1    NEWS_AND_MAGAZINES    173       8352018           48277.560694   
26              FINANCE    243      11631014           47864.255144   
29               COMICS     50       2342284           46845.680000   
23            LIFESTYLE    214       8912231           41645.939252   
14  MAPS_AND_NAVIGATION     92       3520106           38262.021739   
9                DATING    109       3621923           33228.651376   
25       HOUSE_AND_HOME     61       1929707           31634.540984   
0              BUSINESS    164       4931593           30070.689024   
17       ART_AND_DESIGN     49       1192103           24328.632653   
27            PARENTING     48        949667           19784.729167   
22    AUTO_AND_VEHICLES     59       1159320           19649.491525   
3    LIBRARIES_AND_DEMO     61        906493           14860.540984   
24               BEAUTY     37        395989           10702.405405   
13              MEDICAL    135       1166193            8638.466667   
10               EVENTS     33        160777            4872.030303   

    % of Total Apps  
8          0.111185  
30         0.025594  
5          0.029915  
16         0.012631  
28         0.027588  
4          0.034735  
32         0.018115  
15         0.009972  
21         0.024929  
7          0.032408  
12         0.014459  
20         0.194781  
2          0.034070  
31         0.021605  
11         0.031079  
19         0.017284  
6          0.024597  
18         0.081104  
1          0.028752  
26         0.040386  
29         0.008310  
23         0.035566  
14         0.015290  
9          0.018115  
25         0.010138  
0          0.027256  
17         0.008144  
27         0.007977  
22         0.009806  
3          0.010138  
24         0.006149  
13         0.022436  
10         0.005484

The Apple Data:

In [423]:
new_apple = pd.DataFrame(categories_three,columns=['Category'])
new_apple['Count'] = new_apple['Category'].map(a_categ_count)
new_apple['# of Ratings'] = new_apple['Category'].map(a_categ_inst_dict)
new_apple['Avg # Ratings Per App'] = (new_apple['# of Ratings'] / new_apple['Count'])
new_apple['% of Total Apps'] = (new_apple['Count'] / new_apple['Count'].sum())
new_apple.sort_values('% of Total Apps',ascending=False)

Category  Count  # of Ratings  Avg # Ratings Per App  \
21              Games   1274      14150897           11107.454474   
13      Entertainment    191       2357265           12341.701571   
0       Photo & Video    121       1539030           12719.256198   
14  Social Networking     85       1432963           16858.388235   
19          Education     75        538759            7183.453333   
18           Shopping     65       1180422           18160.338462   
8           Utilities     55        775747           14104.490909   
3               Music     51        694758           13622.705882   
2        Productivity     45        757065           16823.666667   
6    Health & Fitness     44        495306           11256.954545   
10             Sports     43        658896           15323.162791   
9           Lifestyle     38        354263            9322.710526   
15             Travel     30        313907           10463.566667   
16               News     28        426322           15225.785714   
11            Finance     22        427138           19415.363636   
4        Food & Drink     17        304090           17887.647059   
17           Business     14        127245            9088.928571   
5           Reference     12        162969           13580.750000   
7             Weather     10        276135           27613.500000   
22               Book      8        304534           38066.750000   
12           Catalogs      4         16016            4004.000000   
1          Navigation      3         16580            5526.666667   
20            Medical      3          3643            1214.333333   

    % of Total Apps  
21         0.569258  
13         0.085344  
0          0.054066  
14         0.037980  
19         0.033512  
18         0.029044  
8          0.024576  
3          0.022788  
2          0.020107  
6          0.019660  
10         0.019214  
9          0.016979  
15         0.013405  
16         0.012511  
11         0.009830  
4          0.007596  
17         0.006256  
5          0.005362  
7          0.004468  
22         0.003575  
12         0.001787  
1          0.001340  
20         0.001340

In [424]:
new_apple.sort_values('Avg # Ratings Per App',ascending=False)

Category  Count  # of Ratings  Avg # Ratings Per App  \
22               Book      8        304534           38066.750000   
7             Weather     10        276135           27613.500000   
11            Finance     22        427138           19415.363636   
18           Shopping     65       1180422           18160.338462   
4        Food & Drink     17        304090           17887.647059   
14  Social Networking     85       1432963           16858.388235   
2        Productivity     45        757065           16823.666667   
10             Sports     43        658896           15323.162791   
16               News     28        426322           15225.785714   
8           Utilities     55        775747           14104.490909   
3               Music     51        694758           13622.705882   
5           Reference     12        162969           13580.750000   
0       Photo & Video    121       1539030           12719.256198   
13      Entertainment    191       2357265           12341.701571   
6    Health & Fitness     44        495306           11256.954545   
21              Games   1274      14150897           11107.454474   
15             Travel     30        313907           10463.566667   
9           Lifestyle     38        354263            9322.710526   
17           Business     14        127245            9088.928571   
19          Education     75        538759            7183.453333   
1          Navigation      3         16580            5526.666667   
12           Catalogs      4         16016            4004.000000   
20            Medical      3          3643            1214.333333   

    % of Total Apps  
22         0.003575  
7          0.004468  
11         0.009830  
18         0.029044  
4          0.007596  
14         0.037980  
2          0.020107  
10         0.019214  
16         0.012511  
8          0.024576  
3          0.022788  
5          0.005362  
0          0.054066  
13         0.085344  
6          0.019660  
21         0.569258  
15         0.013405  
9          0.016979  
17         0.006256  
19         0.033512  
1          0.001340  
12         0.001787  
20         0.001340

## Conclusion

Questions and ideas to consider when developing a new app:

- Filtering out the extremely popular apps gives a clearer picture of how to look at which categories perform the best on each platform.  Minus a revolutionary idea, it probably wouldn't make sense to go up against an app in one of these categories.  

- Cross-checking the categories which are the most popular with the ones that have the most reviews per app (Games on Google and Social on Apple) would show that these categories would be the safer bet.  

- The app development team would have to make a decision, based on app idea, as to finding an area that is not too saturated (such as Family on Google) vs an area that is popular but still has some potential (such as Books on ios).  
- Does the app idea cost continual money to maintain? ie - a paid API to a sports betting site

- Would it be possible to make an app focused on a potentially popular category based on an exisiting popular idea?  

With all of these taken into consideration the data suggests that something in the Books category may prove lucrative - it is a well reviewed category but still has room for growth.  If users were able to interact in a social setting and discuss books, review/recommend books, and maybe even participate in games related to books, I think that something like this would check a lot of the boxes for the popular categories of apps without infringing on the existing apps.  Most of the data would be user-generated and new content is consistently being generated.  